### PR TITLE
Fix various nunit issues

### PR DESCRIPTION
**Pull request by smoogipoo**
* No more static GameHost/AudioManager instances.
* GlyphStore doesn't hard-crash when it fails to load glyphs. Instead, it outputs to the log. This is desirable later on when we want to run rulesets without resources (nunit).
* Fixes a case where Dispose can be run earlier than the last update frame, causing all threads to be stopped without the window closing.
  * This was the cause of nunit tests sometimes hanging.

### CODE REVIEW

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -568,9 +568,12 @@ protected virtual void Dispose(bool disposing)
         {
             if (isDisposed)
                 return;
-
             isDisposed = true;
-            stopAllThreads();
+
+            Exit();
+            while (!exitCompleted && isRunning)
+                InputThread.RunUpdate();
```
Comment (peppy) [inline osu.Framework/Platform/GameHost.cs]: probably needs a comment as to why this is being done (and why the `InputThread`)

**Comment by smoogipoo** [comment]
Gonna run this a few times on CI while testing locally also...

**Comment by smoogipoo** [comment]
Am now running a few iterations of the osu!-side tests locally. Will remove the blocked tag if all good.

**Comment by smoogipoo** [comment]
Have run this 16 times without failing. Removing blocked tag.

**Pull request merged**